### ISSUE TITLE
Trying to upgrade Digitjal Ocean Droplet from beta.15 to beta.18 gives ENOTEMPTY error

### ISSUE THREAD

**Issue by thepilot21** [saltcorn/saltcorn#2717]
**Describe the bug**
I am using the Digital Ocean droplet version of Saltcorn. When i installed this a couple of weeks ago I was able to upgrade the oldish version in the droplet to beta.15 with no problem. However, now I am trying to upgrade from beta.15 to beta.17 or beta.18 and I get:-

Starting upgrade, please wait...
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /home/saltcorn/.local/lib/node_modules/@saltcorn/cli
npm ERR! dest /home/saltcorn/.local/lib/node_modules/@saltcorn/.cli-CWg9jX9y
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rename '/home/saltcorn/.local/lib/node_modules/@saltcorn/cli' -> '/home/saltcorn/.local/lib/node_modules/@saltcorn/.cli-CWg9jX9y'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/saltcorn/.npm/_logs/2024-10-11T09_42_08_559Z-debug-0.log
Upgrade done (if it was available) with code 217.

Press the BACK button in your browser, then RELOAD the page.

When i go back and refresh I am still on beta.15

I noted an item in the wiki with this very error but that seemed to be related to an old node version so I am unsure if I should take that same action (which I think was to delete the node_modules folder and resinstall) on the droplet.

**To Reproduce**
As above

**Expected behavior**
Should upgrade to beta.18

**Comment by christianhugoch** [comment]
Hi, I think you can just delete the folder, even with a newer node version.

But we've added another way to do this, too. When you click the 'choose version' button you should see a dialog where you can select the newest version, and a checkbox to remove the old node_modules directory. Can you try to upgrade with the checkbox clicked

**Comment by thepilot21** [comment]
Actually I did try that earlier on today and it appeared to totaly corrupt saltcorn on my droplet which I could not recover. Had some very strange looking stuff in the saltcorn directory - longish strings which looked like they might be corrupted scripts. Fortunately I had a fairly recent backup so I recreated my droplet from the saltcorn droplet image (which  is around 0.9....) and then did an upgrade (the only option available) and that ran smoothly bringing the image up to beta18. I suspect someone still needs to do some work on the droplet image but for the moment my problem is solved.
